modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/StatisticsGatherer.java

```diff
@@ -152,14 +203,25 @@ public LocalStatisticsGatheringContext gatherLocalObjectsStatisticsAsync(
             if (log.isDebugEnabled())
                 log.debug("Cancel previous statistic gathering for [key=" + key + ']');
 
-            oldCtx.futureGather().cancel(false);
+            oldCtx.cancel();
+        }
+
+        if (!active) {
+            newCtx.cancel();
+            gatheringInProgress.remove(key);
```

Comment:
Do you read 'active' flag and remove key under busylock?

Reply:
Not here, it's just a fast way to skip scheduling if active=false.